src/EFCore.Cosmos/Metadata/Internal/JsonIdDefinition.cs

```
@@ -154,12 +155,18 @@ protected virtual void AppendString(StringBuilder builder, object? propertyValue
     ///     any release. You should only use it directly in your code with extreme caution and knowing that
     ///     doing so can result in application failures when updating to a new Entity Framework Core release.
     /// </summary>
-    protected virtual StringBuilder AppendEscape(StringBuilder builder, string stringValue)
+    protected virtual StringBuilder AppendEscape(StringBuilder builder, string stringValue, bool singleValue)
     {
         var startingIndex = builder.Length;
-        return builder.Append(stringValue)
-            // We need this to avoid collisions with the value separator
-            .Replace("|", "^|", startingIndex, builder.Length - startingIndex)
+        builder = builder.Append(stringValue);
+
+        if (!singleValue)
+        {
+            // We need this to avoid collisions with the value separator, but only when the key has multiple values.
+            builder = builder.Replace("|", "^|", startingIndex, builder.Length - startingIndex);
+        }
+
+        return builder
```

We should just let Cosmos throw for the invalid characters in single value mode to preserve the 1:1 doctrine.